boolean.go

```diff
@@ -162,6 +162,99 @@ func (b *Boolean) Equal(value bool) *Boolean {
 	return b.IsEqual(value)
 }
 
+// InList succeeds if boolean is listed by given [values...].
+//
+// Example:
+//
+//	boolean := NewBoolean(t, true)
+//	boolean.InList(true, false)
+func (b *Boolean) InList(values ...bool) *Boolean {
+	opChain := b.chain.enter("InList()")
+	defer opChain.leave()
+
+	if opChain.failed() {
+		return b
+	}
+
+	if len(values) == 0 {
+		opChain.fail(AssertionFailure{
+			Type: AssertUsage,
+			Errors: []error{
+				errors.New("unexpected empty list argument"),
+			},
+		})
+
+		return b
+	}
+
+	for _, v := range values {
+		if b.value == v {
+			return b
+		}
+	}
+
+	opChain.fail(AssertionFailure{
+		Type:     AssertBelongs,
+		Actual:   &AssertionValue{b.value},
+		Expected: &AssertionValue{AssertionList(boolList(values))},
+		Errors: []error{
+			errors.New("expected: boolean is listed"),
+		},
+	})
+
+	return b
```

Comment:
Nitpick: usually httpexpect methods use the following pattern:

* method can have early returns in case of failures
* there is only one successful return, at the end of the method

You used exactly this layout in Array.InList, but in Boolean.InList you used the opposite layout: early return for success, failure in the end of the method.

I suggest to refactor all InList to follow the same layout as in Array.InList, for consistency.

Reply:
Thanks for the review.

Previously, I tried to use early returns but I had a doubt since types like `Array` receives `interface{}` as param, so I tried to iterate everything first to check each of the params and then decided whether it's succeeded.

I will take this as a guideline
> - method can have early returns in case of failures
> - there is only one successful return, at the end of the method

Reply:
`InList` uses `isListed` and breaks early in each type. While `NotInList` add failed step and breaks early if same.

Let me know if I miss something there, @gavv.